src/Algebra/Construct/DirectSum.agda

```diff
@@ -0,0 +1,181 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Instances of algebraic structures made by taking two other instances
+-- A and B, and having elements of the new instance be pairs |A| × |B|.
+-- In mathematics, this would usually be written A ⊕ B.
+--
+-- From monoids up, these new instances are biproducts – i.e, both
+-- products and coproducts of the relevant category.
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Algebra.Construct.DirectSum where
```

Comment:
I would probably be tempted to call this either `Algebra.Construct.Product` or `DirectProduct` rather than `DirectSum`. [As I understand it](https://mathworld.wolfram.com/GroupDirectSum.html), direct sums are a generalisation of direct products which only differs when it is a product over an infinite number of objects. 

This would also be more consistent with the `Sum` and `Product` terminology used elsewhere in the library.

Reply:
Right, upon reading a bit more, my terminology is totally off here. Probably “biproduct” is the better description for structures with *commutative* addition (commutative monoids, Abelian groups, semirings, rings), in which case the comment at the top of the file is true. Otherwise, it seems that “direct product” or just “product” is right (and it really is just a product in the relevant category).

Concretely, as a module name, `DirectProduct` is probably good enough to cover both.